server/src/weaverbird/backends/pypika_translator/translators/base.py

```diff
@@ -535,23 +533,52 @@ def _get_single_condition_criterion(
                     return column_field.notin(condition.value)
 
             case MatchCondition():
+                compliant_regex = _compliant_regex(condition.value)
 
                 if condition.operator == "matches":
                     match self.REGEXP_OP:
                         case RegexOp.REGEXP:
                             return column_field.regexp(condition.value)
                         case RegexOp.SIMILAR_TO:
+
```

Comment:
I feel like there should be an abstract method returning the match operator: Here, we're guessing what the child class needs and we have cases dependent on child classes (like on l.544, 561 or 572). I believe that it would lead to a better separation of concerns and easier to read code (and probably less LoC if we eliminate some cases).

We can keep this as is for this PR and do the refactoring later though